### PR TITLE
CNetworkPosix: implement ping using sockets instead of system call to ping

### PR DESCRIPTION
**Pull request by lrusak**
This removes the system call to use the ping command and implements ping directly using UDP sockets.

This could possibly be used on all posix platforms if I used `poll` instead of `epoll`. I might be persuaded to do that if desired.

I've also added a simple test to test the `CNetworkBase::PingHost` command.

The logging output is similar to the ping command output, for example:
```
2023-10-02 23:48:25.843 T:423929    info <general>: WakeOnAccess [192.168.1.108] triggered by accessing : nfs://192.168.1.108/mnt/tank/concerts/folder.jpg
2023-10-02 23:48:25.844 T:423929   debug <general>: PING 192.168.1.108: icmp_seq=0 ttl=64 time=0.430 ms
2023-10-02 23:48:25.844 T:423929    info <general>: WakeOnAccess success exit, server already running
```

### CODE REVIEW

**Comment by enen92** [comment]
> This could possibly be used on all posix platforms if I used `poll` instead of `epoll`. I might be persuaded to do that if desired.
> 

That would be pretty great (would allow us to drop a similar implementation in macos) but maybe asking for too much :)
+1 (but can also be done later)

File: xbmc/platform/posix/network/NetworkPosix.cpp [hunk 1]
```diff
@@ -148,3 +192,89 @@ CNetworkInterface* CNetworkPosix::GetFirstConnectedInterface()
...
+bool CNetworkPosix::PingHost(unsigned long remote_ip, unsigned int timeout_ms)
+{
+  CFileHandle fd(socket(AF_INET, SOCK_DGRAM | SOCK_NONBLOCK | SOCK_CLOEXEC, IPPROTO_ICMP));
+  if (!fd)
+  {
+    CLog::Log(LOGERROR, "socket failed: {} ({})", strerror(errno), errno);
+    return false;
+  }
+
+  int ret = setsockopt(fd, SOL_IP, IP_TTL, &TTL, sizeof(TTL));
```
Comment (enen92) [inline xbmc/platform/posix/network/NetworkPosix.cpp:205]: From https://man7.org/linux/man-pages/man7/ip.7.html:
>> Some BSD sockets implementations also provide an IP_RECVTTL option, but an ancillary message with type IP_RECVTTL is passed with the incoming packet.  This is different from the IP_TTL option used in Linux.
Using the SOL_IP socket options level isn't portable; BSD-based stacks use the IPPROTO_IP level.

It seems Linux is not posix compliant here, not sure at which extent you can keep a common implementation for all. Maybe you'll have to reimplement for Linux (https://stackoverflow.com/questions/16165542/is-it-possible-to-read-the-ttl-ip-header-field-when-receiving-udp-packets)

Comment (lrusak) [reply xbmc/platform/posix/network/NetworkPosix.cpp]: I think I leave it as an exercise for others to implement it on their platforms :)

**Comment by lrusak** [comment]
I'm not really sure what to do about the windows include :unamused:

**Comment by garbear** [comment]
Is the include only needed for SOCKET or for something else? CreateTCPServerSocket() could be moved to a new file

**Comment by lrusak** [comment]
> Is the include only needed for SOCKET or for something else? CreateTCPServerSocket() could be moved to a new file

Yea just for `SOCKET`. I think a proper solution would be avoiding the use of a windows type and use a custom implementation instead.

**Comment by lrusak** [comment]
I just added a commit to add `<`winsock2.h>` to PlatformDef.h for windows. Seems simple enough for now.

**Pull request merged**